### PR TITLE
[WIP] Clickhouse-benchmark comparison mode

### PR DESCRIPTION
**Pull request by dimarub2000**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):
#6220 Implemented

### CODE REVIEW

**Comment by dimarub2000** [comment]
Was thinking about 

> You can also implement some statistical tests for "is median (or mean) of one distribution is less than another" and show the results of these tests on the fly.

Looks like query_time_percentiles are enough for comparison.

File: dbms/programs/benchmark/Benchmark.cpp [hunk 1]
```diff
@@ -160,15 +184,189 @@ class Benchmark : public Poco::Util::Application
...
+            double squares_sum = 0;
+
+            void add(double seconds)
+            {
+                ++cnt;
+                sum += seconds;
+                squares_sum += seconds * seconds;
+            }
+
+            double avg()
```
Comment (alexey-milovidov) [inline dbms/programs/benchmark/Benchmark.cpp]: const

File: dbms/programs/benchmark/Benchmark.cpp [hunk 2]
```diff
@@ -160,15 +184,189 @@ class Benchmark : public Poco::Util::Application
...
+            {
+                return sum / cnt;
+            }
+
+            double var()
```
Comment (alexey-milovidov) [inline dbms/programs/benchmark/Benchmark.cpp]: const

File: dbms/programs/benchmark/Benchmark.cpp [hunk 3]
```diff
@@ -622,7 +623,7 @@ class Benchmark : public Poco::Util::Application
 
         for (size_t i = 1; i <= infos.size(); ++i)
         {
-            auto info = infos[i - 1];
+            auto & info = infos[i - 1];
```
Comment (alexey-milovidov) [inline dbms/programs/benchmark/Benchmark.cpp]: const?

File: dbms/programs/benchmark/Benchmark.cpp [hunk 4]
```diff
@@ -552,17 +551,18 @@ class Benchmark : public Poco::Util::Application
         std::lock_guard lock(mutex);
 
         std::cerr << "\n";
-        size_t info_counter = 1;
-        for (auto & info : infos)
+        for (size_t i = 1; i <= infos.size(); ++i)
         {
+            auto & info = infos[i - 1];
```
Comment (alexey-milovidov) [inline dbms/programs/benchmark/Benchmark.cpp]: const?

**Comment by alexey-milovidov** [comment]
Please prepare for merge.

**Comment by dimarub2000** [comment]
> Please prepare for merge.

Added a T_test.h file like in performance-test PR, and made options --host --port similar to performance-test PR, think it is ready now.

**Comment by alexey-milovidov** [comment]
> UPD: Not sure if i should fix style of a huge table (Style check — Found 101 errors)

Sure, you have to.

**Comment by alexey-milovidov** [comment]
TODO: rename T_test to StudentTTest, move code to .cpp.

**Pull request merged**